lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -5296,6 +5296,9 @@ class GenericSignatureBuilder::InferRequirementsWalker : public TypeWalker {
     auto decl = ty->getAnyNominal();
     if (!decl) return Action::Continue;
 
+    if (!decl->hasComputedGenericSignature())
```

Comment:
Why is this necessary? I would expect that kicking off a request from the GSB is totally the right thing to do

Reply:
This is a weird compromise.  The GSB does cycle detection, and so does the request evaluator.  We wind up double-diagnosing because of it in a few tests.

Reply:
We should clean this up later. Can you file a radar?

Reply:
rdar://55263708.  Will attach in a FIXME as well.